dash_vtk/utils/vtk.py

```diff
@@ -11,184 +13,217 @@
 
 # Numpy to JS TypedArray
 to_js_type = {
-    'int8': 'Int8Array',
-    'uint8': 'Uint8Array',
-    'int16': 'Int16Array',
-    'uint16': 'Uint16Array',
-    'int32': 'Int32Array',
-    'uint32': 'Uint32Array',
-    'int64': 'Int32Array',
-    'uint64': 'Uint32Array',
-    'float32': 'Float32Array',
-    'float64': 'Float64Array',
+    "int8": "Int8Array",
+    "uint8": "Uint8Array",
+    "int16": "Int16Array",
+    "uint16": "Uint16Array",
+    "int32": "Int32Array",
+    "uint32": "Uint32Array",
+    "int64": "Int32Array",
+    "uint64": "Uint32Array",
+    "float32": "Float32Array",
+    "float64": "Float64Array",
 }
 
 
+def b64_encode_numpy(obj):
+    # Convert 1D numpy arrays with numeric types to memoryviews with
+    # datatype and shape metadata.
+    dtype = obj.dtype
+    if (
+        dtype.kind in ["u", "i", "f"]
+        and str(dtype) != "int64"
+        and str(dtype) != "uint64"
+    ):
+        # We have a numpy array that is compatible with JavaScript typed
+        # arrays
+        buffer = base64.b64encode(memoryview(obj.ravel(order="C"))).decode("utf-8")
+        return {"bvals": buffer, "dtype": str(dtype), "shape": obj.shape}
```

Comment:
My understanding is that the original code was meant for 1D numpy array. I don't think that creates any problem for obj.shape (since a python tuple should be easy to convert to JS), but i'm curious whether there will be needs to do any modification on the JS side for that to work...

Reply:
On the vtk side we expect 1D, we don't care about the shape... Just kept it for @nicolaskruchten . ;-)

Reply:
sure!